s26856-DawidCybuch/zadanie_3.txt

```diff
@@ -0,0 +1,236 @@
+Zadanie 0.1
+Przejdź do swojego katalogu domowego
+
+Używam komendy cd ~
+I znajduje się w swoim folderze domowym
+
+Zadanie 0.2
+Wydaj komendę: ls -a
+
+Wyświetliło mi wszystkie kadtalogi, nawet te ukryte
+
+Zadanie 0.3
+Zobacz ile plików wypisało.
+
+Wypisało mi 7 plików oraz 17 folderów
+
+Zadanie 0.4
+Teraz wydaj komendę: ls -a | grep D
+HINT: Jeśli nie masz żadnego pliku czy katalogu z literką D to wybierz inną literkę, która już będzie w nazwie jakiegoś pliku czy katalogu.
+
+W moim przypadku została użyta komenda ls -a | grep U
+
+Zadanie 0.5
+Zobacz ile teraz jest wyników. Co się stało?
+Otóż program grep służy do wyszukiwania wierszy w pliku lub strumieniu wejściowym, które pasują do wzorca. Tu podano wzorzec jako "D".
+
+Zostały wypisane wszystkie pliki oraz foldery zawierające, w moim przypadku litere U 
+Jest ich 5
+
+Zadanie 0.6
+A teraz wykonaj komendę: ls -a | grep D > lista_plikow_z_literka_d.txt
+
+W moim przypadku została użyta komenda ls -a | grep U > lista_plików_z_literka_u.txt
+Z racji tego że użyłem tutaj dużego U zapisane są tu tylko 2 plikitee
+
+Zadanie 0.7
+Zobacz czy utworzył sie jakiś plik?
+Jaka jest jego treść?
+Co znaczy | oraz co znaczy > ?
+
+Utworzyłem plik o nazwie lista_plikow_z_literka_u.txt
+Trescia pliku sa pliki z literka U
+| oznacza przekierowanie wyjscia jendego programu na wejscie drugiego
+> oznacza przekierowanie wyjścia z programu do pliku
+
+Zadanie 0.8
+A teraz wykonaj komendę: ls -a | grep D | tee lista_plikow_z_literka_d_2.txt
+Zobacz czy plik się utworzył?
+Jaka jest treść względem poprzedniej próby bez programu tee?
+Co robi program tee?
+
+Została użyta komenda ls -a | grep u | tee lista_plikow_z_literka_u_2.txt
+Plik został utworzony
+Wcześniej plik został tylko utworzony
+Z racji tego że użyłem tee, została pokazana zawartość pliku
+Komenda tee służy do wyświetlania treści pliku
+
+Zadanie 1
+
+Program ps służy do wyświetlania listy procesów. 
+
+Zadania:
+
+A: Zobacz co się stanie jeśli wpiszemy w terminalu:
+
+ps
+
+ps -a
+
+ps x
+
+ps axu
+INFO: tu kolejność parametrów a, x, u nie ma znaczenie a jedynie fakt, że są podane. 
+Dowolna kolejność powinna dać ten sam wynik.
+
+Jak myślisz, co oznacza znak zapytania w kolumnie 2?
+Nie wiesz? A gdzie może być wyjaśnienie?
+
+B: Jak wykonasz poniższe 2 zadania?
+
+Wyświetl wszystkie procesy bash
+
+Wyświetl wszystkie procesy należące do użytkownika root
+
+A.1
+
+Wyświetla listę procesów użytkownika
+
+A.2
+
+Wyświetla lsitę procesów, które nie są powiązane z terminalem
+
+A.3
+
+Wyświetla wszystkie procesy użytkownika, w tym te ukryte
+
+A.4
+
+
+Wypisuje wszystkie możliwe procesy
+Literka x oznacza należące do użytkowników
+Gdybym nie wiedział wpisuje komende man ps i szukam co oznacza x
+
+B.1
+
+Używam komeny ps axu | grep bash
+Z racji tego że robię to w godzinach nocnych wyświetla mi tylko dwa procesy
+Ale gdybym robił to w dzień, pokazało by mi ich znacznie więcej
+
+B.2
+
+Wykonałem komende ps axu | grep ^root
```

Comment:
A jeśli user o loginie `barnabash` będzie miał uruchomiony proces o nazwie `i_am_groot` zostanie wypisany?
Będzie zatem to user root czy nie?